pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -59,13 +65,28 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(Schema.class);
 
   private String _schemaName;
+
+  // upsert related config
+  // primary key refers to the column name that is the primary key of this upsert table
+  private String _primaryKey;
+  // offset key refers to the column name that we are going to store the offset value to
+  private String _offsetKey;
```

Comment:
do we need all these things in Schema? What if we create a UpsertConfig in TableConfig to provide all the necessary information.

Reply:
definitely would be better this way, will submit a change later to update this